<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Finally Clause</title>
</head>

<body>
    <h1>how are you</h1>
    <script>
        // The finally Clause
        // The try catch construct may have one more code Clause: finally.
        // If it exists it runs in all cases.
        // try {
        //     let a = 0;
        //     console.log(program);
        // } catch (error) {
        //     console.log('This is an error.');
        //     fds
        // }
        // finally{ // This will be executed even if catch contains an error.
        //     console.log('I am a good boy.');
        // }
        // console.log('I am a very good boy'); // This will not be executed if catch contains an error.
        // Uses of finally
        // 1. Close the file.
        // 2. Exit the Loop.
        // 3. Write to the log file.

        // Above syntax in function which returns
        const f = () => {
            try {
                let a = 0;
                console.log("Program runs successfully.");
                return 'Hello'
                877
            } catch (error) {
                console.log('This is an error.');
                fds
            }
            finally { // This will be executed even if catch contains an error.
                console.log('I am a good boy.');
            }
            console.log('I am a very good boy'); // This will not be executed if catch contains an error.
        }
        // // f();
        // console.log(f());
    </script>
</body>

</html>